[flaky test][exporter/elasticsearch] TestExporterLogs tests flaky on Windows

### Component(s)

exporter/elasticsearch

### Describe the issue you're reporting

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/12815700710/job/35735010165#step:8:528

```
=== FAIL: . TestExporterLogs/no_retry/fail_item/retry.enabled_is_false (0.28s)
    exporter_test.go:577: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/elasticsearchexporter/exporter_test.go:577
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestExporterLogs/no_retry/fail_item/retry.enabled_is_false
make[2]: *** [../../Makefile.Common:134: test] Error 1

=== FAIL: . TestExporterLogs/no_retry/fail_item (0.00s)

=== FAIL: . TestExporterLogs/no_retry/fail_http_request/retry.enabled_is_false (0.27s)
    exporter_test.go:577: 
make[1]: *** [Makefile:200: exporter/elasticsearchexporter] Error 2
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/elasticsearchexporter/exporter_test.go:577
make[1]: *** Waiting for unfinished jobs....
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestExporterLogs/no_retry/fail_http_request/retry.enabled_is_false

=== FAIL: . TestExporterLogs/no_retry/fail_http_request (0.00s)

=== FAIL: . TestExporterLogs/no_retry (0.00s)

=== FAIL: . TestExporterLogs (3.46s)

=== FAIL: . TestExporterLogs/no_retry/fail_item/retry.enabled_is_false (re-run 1) (0.40s)
    exporter_test.go:577: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/elasticsearchexporter/exporter_test.go:577
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestExporterLogs/no_retry/fail_item/retry.enabled_is_false

=== FAIL: . TestExporterLogs/no_retry/fail_item (re-run 1) (0.00s)

=== FAIL: . TestExporterLogs/no_retry (re-run 1) (0.00s)

=== FAIL: . TestExporterLogs (re-run 1) (0.41s)
```

--- a/exporter/elasticsearchexporter/exporter_test.go
+++ b/exporter/elasticsearchexporter/exporter_test.go
@@ -546,7 +546,7 @@ func TestExporterLogs(t *testing.T) {
 			},
 		}
 
-		handlers := map[string]func(attempts *atomic.Int64) bulkHandler{
+		handlers := map[string]func(*atomic.Int64) bulkHandler{
 			"fail http request": func(attempts *atomic.Int64) bulkHandler {
 				return func([]itemRequest) ([]itemResponse, error) {
 					attempts.Add(1)
@@ -573,8 +573,9 @@ func TestExporterLogs(t *testing.T) {
 						exporter := newTestLogsExporter(t, server.URL, configurer)
 						mustSendLogRecords(t, exporter, plog.NewLogRecord())
 
-						time.Sleep(200 * time.Millisecond)
-						assert.Equal(t, int64(1), attempts.Load())
+						assert.Eventually(t, func() bool {
+							return attempts.Load() == 1
+						}, 5*time.Second, 20*time.Millisecond)
 					})
 				}
 			})

